fastlane_core/lib/fastlane_core/print_table.rb

```diff
@@ -34,16 +40,68 @@ def print_values(config: nil, title: nil, hide_keys: [], mask_keys: [])
         return params
       end
 
-      def limit_row_size(rows, max_length = 100)
+      def colorize_array(array, colors)
+        value = ""
+        array.each do  |l|
+          colored_line = l
+          colored_line = "#{colors.first[0]}#{l}#{colors.last[0]}" if colors.length > 0
+          value << colored_line
+          value << "\n"
+        end
+        return value
+      end
+
+      def should_transform?
+        !FastlaneCore::Env.truthy?("FL_SKIP_TABLE_TRANSFORM")
+      end
+
+      def transform_output(rows, transform: :newline)
+        return rows unless should_transform?
+
         require 'fastlane_core/string_filters'
+        require 'tty-screen'
 
-        max_key_length = rows.map { |e| e[0].length }.max || 0
-        max_allowed_value_length = max_length - max_key_length - 7
-        rows.map do |e|
-          value = e[1]
-          value = value.to_s.truncate(max_allowed_value_length) unless [true, false].include?(value)
-          [e[0], value]
+        number_of_cols = TTY::Screen.width
+
+        col_count = rows.map(&:length).first || 1
+
+        # -4 per column - as tt adds "| " and " |"
+        terminal_table_padding = 4
+        max_length = number_of_cols - (col_count * terminal_table_padding)
+
+        max_value_length = (max_length / col_count)
+
+        return_array = rows.map do |row|
+          row.map do |col|
+            if col.nil?
+              col # we want to keep the nil and not convert it to a string
```

Comment:
Agreed about extracting a method here. You could also consider starting with

```ruby
return col if col.nil?
```

to allow you to remove the outer if/else.